core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java

```diff
@@ -161,6 +166,10 @@ public FlowIdentifier stopInstances(Stack stack, Collection<String> instanceIds,
         if (instanceIdsByHostgroupMap.size() > 1) {
             throw new BadRequestException("Downscale via Instance Stop cannot process more than one host group");
         }
+        updateNodeCountValidator.validateInstanceGroup(stack, hostGroup);
+        if (!hostGroup.equalsIgnoreCase(new ArrayList<>(instanceIdsByHostgroupMap.keySet()).get(0))) {
```

Comment:
Can be replaced by
`!hostGroup.equalsIgnoreCase(instanceIdsByHostgroupMap.keySet().iterator().next())`

Reply:
Ack, will update this